R/list-flatten.R

```diff
@@ -45,11 +46,20 @@ list_flatten <- function(
   ) {
   vec_check_list(x)
 
-  x <- map_if(x, vec_is_list, identity, .else = list)
-  vec_unchop(
-    x,
+  # Take the proxy as we restore on exit
+  proxy <- vec_proxy(x)
```

Comment:
Currently this isn't necessary, as lists are always their own proxies, but it's good practice to only call restore on proxies, and also in the future we might want to have proxied lists e.g. to support special bioc types.